docs/source/clickhouse/schema_design.rst

```diff
@@ -73,3 +74,78 @@ Those styles of queries require a separate column to search.
            attributes. After the table goes through a full TTL period and the API/storage definition
            is changed to serve the values from the top-level field, message processors should be changed
            to stop writing the data in duplicate places.
+
+
+Aggregate Tables and Materialization
+------------------------------------
+
+A common use case for ClickHouse and Snuba is to ingest raw data and automatically
+roll it up to aggregate values (keyed by a custom set of dimensions). This lets
+a dataset owner simplify their write logic while getting the query benefits of
+rows that are pre-aggregated. This is done with what we'll call a raw table
+(the table the consumer writes to), an aggregate table (the table the API reads from)
+and a materialized view (which describes how the data should be transformed from
+raw to aggregate).
+
+`Sample usage of a materialized view/aggregate table from the official ClickHouse Documentation <https://clickhouse.com/docs/en/engines/table-engines/mergetree-family/aggregatingmergetree#example-of-an-aggregated-materialized-view>`_.
+Note that contrary to this example, the aggregate table definition in Snuba is
+always separate from the materialized view definition (which is just a ClickHouse SQL
+transformation, similar to a PostgreSQL trigger).
+
+In general, Snuba follows the naming conventions here:
+
+* (``widgets_raw_local``, ``widgets_raw_dist``) for the raw (local, distributed) tables
+* ``widgets_aggregation_mv`` for the materialized view (this only exists on storage nodes)
+* (``widgets_aggregated_local``, ``widgets_aggregated_dist``) for the roll-up/aggregated (local, distributed) tables
+
+Materialized views are immutable so it's normal to have multiple versions of
+``widgets_aggregation_mv`` when behavior is updated, with suffixes like
+``widgets_aggregation_mv_v1``, ``widgets_aggregation_mv_v2``, etc. Migration
+between materialized view versions are described in the next section but in general
+old materialized views should be discarded once they are no longer used.
+
+Schema migrations using materialization_version
```

Comment:
Please deprecate and delete the internal documentation when this is merged so we do not have two independent places that will evolve on their own.